# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/prim_store/picture.rs
# Commit: b7f88e5c537b
# Full Hash: b7f88e5c537bb7d64ba9d4cb58b052650d510638
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1749380
# File Overlap Count: 1
# Description:
#   Bug 1749380 - Improve how WR handles bounding rects for off-screen surfaces r=gfx-reviewers,kvark
#   
#   This patch introduces a number of subtle but important changes
#   to how we deal with off-screen surfaces. The overall goals are:
#   
# ==============================================================================

diff -r 8e9c1ffdeed8 -r b7f88e5c537b gfx/wr/webrender/src/prim_store/picture.rs
--- a/gfx/wr/webrender/src/prim_store/picture.rs	Mon Feb 07 22:02:35 2022 +0000
+++ b/gfx/wr/webrender/src/prim_store/picture.rs	Mon Feb 07 22:14:47 2022 +0000
@@ -79,7 +79,7 @@
     Identity,
 
     // FilterOp
-    Blur(Au, Au),
+    Blur(Au, Au, bool),
     Brightness(Au),
     Contrast(Au),
     Grayscale(Au),
@@ -140,8 +140,8 @@
             }
             Some(PictureCompositeMode::Filter(op)) => {
                 match op {
-                    Filter::Blur(width, height) =>
-                        PictureCompositeKey::Blur(Au::from_f32_px(width), Au::from_f32_px(height)),
+                    Filter::Blur { width, height, should_inflate } =>
+                        PictureCompositeKey::Blur(Au::from_f32_px(width), Au::from_f32_px(height), should_inflate),
                     Filter::Brightness(value) => PictureCompositeKey::Brightness(Au::from_f32_px(value)),
                     Filter::Contrast(value) => PictureCompositeKey::Contrast(Au::from_f32_px(value)),
                     Filter::Grayscale(value) => PictureCompositeKey::Grayscale(Au::from_f32_px(value)),